nomad/job_endpoint_hook_connect.go

```diff
@@ -230,6 +230,29 @@ func injectPort(group *structs.TaskGroup, label string) {
 	})
 }
 
+// groupConnectGuessTaskDriver will scan the tasks in g and try to decide which
+// task driver to use for the default sidecar proxy task definition.
+//
+// If there is at least one podman task and zero docker tasks, use podman.
+// Otherwise default to docker.
+//
+// If the sidecar_task block is set, that takes precedence and this does not apply.
+func groupConnectGuessTaskDriver(g *structs.TaskGroup) string {
+	foundPodman := false
+	for _, task := range g.Tasks {
+		switch task.Driver {
+		case "docker":
+			return "docker"
+		case "podman":
+			foundPodman = true
+		}
+	}
+	if foundPodman {
+		return "podman"
+	}
+	return "docker"
+}
```

Comment:
It's a simple function, so I hate to bikeshed on it 😅 

But I found this logic a bit confusing to grok at first. I think we could instead have a task driver counter map and then something like this so it more explicitly matches the function docstring:

```go
if numTasks["podman"] > 1 && numTasks["docker"] == 0 {
  return "podman"
}

return "docker"
```

Reply:
How about

```go
func groupConnectGuessTaskDriver(g *structs.TaskGroup) string {
    drivers := set.FromFunc(g.Tasks, func(t *structs.Task) string {
        return t.Driver
    })
    if drivers.Contains("podman") && !drivers.Contains("docker") {
        return "podman"
    }
    return "docker"
}
```